Source/JavaScriptCore/parser/Nodes.h

```diff
@@ -906,6 +910,18 @@ namespace JSC {
         ExpressionNode* base() const { return m_base; }
         const Identifier& identifier() const { return m_ident; }
 
+        bool isArgumentsAccess(VM& vm) const final
+        {
+            return m_base->isArguments(vm);
+        }
+
+        bool isArgumentsLengthAccess(VM& vm) const final
+        {
+            return m_base->isResolveNode()
+                && static_cast<ResolveNode*>(m_base)->identifier() == vm.propertyNames->arguments
```

Comment:
Why does this need to check `m_base->isResolveNode()` but `isArgumentsAccess` does not need to do the same check?  Why not just call `m_base->isArguments(vm)` instead?  I see that `isArguments` is a virtual function though.

Reply:
Yes, we should use `m_base->isArguments(vm)`.